cmd/kubeadm/app/master/manifests.go

```diff
@@ -355,7 +355,9 @@ func getAPIServerCommand(cfg *kubeadmapi.MasterConfiguration, selfHosted bool, k
 		defaultArguments["proxy-client-cert-file"] = filepath.Join(cfg.CertificatesDir, kubeadmconstants.FrontProxyClientCertName)
 		defaultArguments["proxy-client-key-file"] = filepath.Join(cfg.CertificatesDir, kubeadmconstants.FrontProxyClientKeyName)
 	}
-	if k8sVersion.AtLeast(kubeadmconstants.MinimumNodeAuthorizerVersion) {
+	// TODO: That we have to exclude v1.8.0-alpha.0 here is based on the way kubernetes branches work
+	// v1.7.0-beta.0 == v1.8.0.alpha.0 but they are sorted/compared differently
+	if k8sVersion.AtLeast(kubeadmconstants.MinimumNodeAuthorizerVersion) && cfg.KubernetesVersion != "v1.8.0-alpha.0" {
```

Comment:
Okay, after looking at the repository version tags, I understand now. Ideally, semver means that 1.8.0-alpha.0 came strictly after anything in 1.7, and our engine handles that correctly, but we have actually applied the version tag to the same commit as 1.7.0-beta.0, so the semver comparison doesn't actually make sense, and 1.7.1-beta.1 actually came after 1.8.0-alpha.0 in the commit history. Yuck.

Since the release cycles for wrapping up 1.7 and pushing forward with 1.8 are overlapping, I wonder if the better solution (instead of special-casing 1.8.0-alpha.0) is to have a minimum 1.8 version to check as well. Someone could still tag 1.8.0-alpha.1 to be a commit that logically occurred before 1.7.0-beta.1, but it should at least be a commit later than 1.8.0-alpha.0. I hope that we at least have strong semver guarantees within each 1.X lineage.

So, I'd recommend changing this from an exact check of 1.8.0-alpha.0 to instead check for >= 1.8.0-alpha.1. That way, we're logically asking "is this greater than the first release that included the feature in the 1.7 or 1.8 series?" instead of saying "1.8.0-alpha.0 is a weird botched special case we have to handle specifically."

If the logic gets complicated, I would also pull it out into a well-named helper like `IsNodeAuthorizerSupported` that we can much more thoroughly test for all of the version edge cases.